### PR TITLE
Replace some optimizers

### PR DESCRIPTION
**Pull request by ArrowM**
Optimizer changes:
- remove AdanIP Adapt
- replace Adan Adapt
- Add AdamW Adapt, Lion Adapt, and SGD Adapt

A couple library updates

A couple linting changes

### CODE REVIEW

**Comment by ArrowM** [comment]
Lion seems to be worse than AdamW at any of the batch sizes people would use for Dreambooth, so I removed it.

Updated optimizers seem similar to 8Bit AdamW in terms of quality. I included some recommended settings in the Optimizer tooltip.
SGD and Adan are a little worse performance-wise.

**Comment by ashleykleynhans** [comment]
Lion is actually my preferred optimizer.

It works well with a learning rate of 1e-7 and weight decay of 0.1.

Can we leave it as an option in addition to your other changes, rather than removing it?

**Comment by FurkanGozukara** [comment]
> Lion

I am using Lion  and it is better

I compared all here : https://youtu.be/sRdtVanSRl4

Why removing something that works?

**Comment by ArrowM** [comment]
I'll un-remove Lion when I get home later
